plugin/forward/health.go

```diff
@@ -14,10 +14,14 @@ import (
 type HealthChecker interface {
 	Check(*Proxy) error
 	SetTLSConfig(*tls.Config)
+	SetRecursionDesired(bool)
 }
 
 // dnsHc is a health checker for a DNS endpoint (DNS, and DoT).
-type dnsHc struct{ c *dns.Client }
+type dnsHc struct {
+	c                *dns.Client
+	recursionDesired bool
+}
 
 // NewHealthChecker returns a new HealthChecker based on transport.
 func NewHealthChecker(trans string) HealthChecker {
```

Comment:
this should probably receive a boolean or an `options` to set the recursion field

Reply:
I added a bool to HealthCheckers signature. Are you ok with this being set to true default in NewProxy()? Or should I also edit the signature of NewProxy() to include this config?